service/dynamodb/expression/operand.go

```diff
@@ -11,109 +9,64 @@ import (
 // (i.e. AND, BETWEEN, EQUALS)
 type SizeBuilder struct {
-	pb PathBuilder
-}
-
-// Expression contains the map of aliases to names/values, representing the
-// ExpressionAttributeNames and ExpressionAttributeValues, that is needed in
-// order for the Expression string to be used in an operation input.
-// (i.e. UpdateItemInput, DeleteItemInput, etc)
-//
-// Example:
-//
-//     // let expr be an instance of Expression{}
-//
-//     deleteInput := dynamodb.DeleteItemInput{
-//       ConditionExpression:       aws.String(expr.Expression),
-//       ExpressionAttributeNames:  expr.Names,
-//       ExpressionAttributeValues: expr.Values,
-//       Key: map[string]*dynamodb.AttributeValue{
-//         "PartitionKey": &dynamodb.AttributeValue{
-//           S: aws.String("SomeKey"),
-//         },
-//       },
-//       TableName: aws.String("SomeTable"),
-//     }
-type Expression struct {
-	Names      map[string]*string
-	Values     map[string]*dynamodb.AttributeValue
-	Expression string
-}
-
-// ExprNode will be the generic nodes that will represent both Operands and
-// Conditions. The purpose of ExprNode is to be able to call an generic
-// recursive function on the top level ExprNode to be able to determine a root
-// node in order to deduplicate name aliases.
-// fmtExpr is a string that has escaped characters to refer to
-// names/values/children which needs to be aliased at runtime in order to avoid
-// duplicate values. The rules are as follows:
-//     $p: Indicates that an alias of a name needs to be inserted. The corresponding
-//         name to be aliased will be in the []names slice.
-//     $v: Indicates that an alias of a value needs to be inserted. The
-//         corresponding value to be aliased will be in the []values slice.
-//     $c: Indicates that the fmtExpr of a child ExprNode needs to be inserted. The
-//         corresponding child node is in the []children slice.
-type ExprNode struct {
-	names    []string
-	values   []dynamodb.AttributeValue
-	children []ExprNode
-	fmtExpr  string
+	nameBuilder NameBuilder
 }
 
 // OperandBuilder represents the idea of Operand which are building blocks to
 // DynamoDB Expressions. OperandBuilders will be children of ConditionBuilders
 // to represent a tree like structure of Expression dependencies. The method
-// BuildOperand() will create an instance of ExprNode, which is an generic
-// representation of both Operands and Conditions. BuildOperand() will mainly
-// be called recursively by the BuildExpression() method call when Expressions
-// are built from ConditionBuilders
+// BuildNode() will create an instance of ExprNode, which is an generic
+// representation of both Operands and Conditions. BuildNode() will mainly be
+// called recursively by the BuildTree() method call when Builder is built from
+// ConditionBuilders
 type OperandBuilder interface {
-	BuildOperand() (ExprNode, error)
+	BuildNode() (ExprNode, error)
```

Comment:
Suggest this be a `BuildOperand` which returns a `Operand` type which could be something like the following.

```go
type Operand struct {
   Name string
   Value dynamodb.AttributeValue
   Expression string
}
```

This will allow you to make `ExprNode` private,